### PR TITLE
Support direct access server option

### PR DESCRIPTION
**Pull request by dplewis**
Closes: https://github.com/parse-community/parse-server/issues/5442

Let me know if I missed anything or wording needs clarifying. I've never added a server config option before.

### CODE REVIEW

**Comment by omairvaiyani** [comment]
I'd love to quantify how much time this will save per request, I suspect it'll be quite noticeable at scale. Nice work.

**Comment by acinader** [comment]
@omairvaiyani the reason that it hasn't been turned on is because the unit tests fail.

@dplewis you could add this to the travis matrix and just make it optional so we can at least see what we're up against?

File: src/Options/docs.js [hunk 1]
```diff
@@ -50,6 +51,7 @@
  * @property {Number} schemaCacheTTL The TTL for caching the schema for optimizing read/write operations. You should put a long TTL when your DB is in production. default to 5000; set 0 to disable.
  * @property {Number} cacheTTL Sets the TTL for the in memory cache (in ms), defaults to 5000 (5 seconds)
  * @property {Number} cacheMaxSize Sets the maximum size for the in memory cache, defaults to 10000
+ * @property {Boolean} directAccess Replace HTTP Interface when using JS SDK in current node runtime
```
[the remark this answers is not in the document]
Comment (dplewis) [reply src/Options/docs.js]: I realized updating Parse Server config definitions isn't documented and isn't run on every PR. I can submit a separate PR for that

File: README.md [hunk 2]
```diff
@@ -245,6 +245,7 @@ The client keys used with Parse are no longer necessary with Parse Server. If yo
 * `auth` - Used to configure support for [3rd party authentication](http://docs.parseplatform.org/parse-server/guide/#oauth-and-3rd-party-authentication).
 * `facebookAppIds` - An array of valid Facebook application IDs that users may authenticate with.
 * `mountPath` - Mount path for the server. Defaults to `/parse`.
+* `directAccess` - Replace HTTP Interface when using JS SDK in current node runtime. Defaults to false.
```
[the remark this answers is not in the document]
Comment (dplewis) [reply README.md]: We should remove this section, organize the definition in alphabetical order and replace with http://parseplatform.org/parse-server/api/3.3.0/ParseServerOptions.html

**Comment by dplewis** [comment]
@acinader I reverted travis.yml because the matrix stakes too long. At least we can see what some of the problems are. Looks like there are issues with CLP, ACL, MasterKey and SessionToken.

https://travis-ci.org/parse-community/parse-server/jobs/529921508

Besides that, how does this look?

File: src/Options/Definitions.js [hunk 3]
```diff
@@ -148,16 +148,16 @@ module.exports.ParseServerOptions = {
...
     default: ['email'],
   },
   protectedFields: {
     env: 'PARSE_SERVER_PROTECTED_FIELDS',
     help:
-      'Personally identifiable information fields in the user table the should be removed for non-authorized users.',
+      'Protected fields that should be treated with extra security when fetching details.',
     action: parsers.objectParser,
-    default: { _User: { '*': ['email'] } },
+    default: [],
```
Comment (acinader) [inline src/Options/Definitions.js]: why the change to the default?

Comment (dplewis) [reply src/Options/Definitions.js]: This is autogenerated. ObjectParser doesn't exist for this json. I can fix this in a separate PR.
